### ISSUE TITLE
react-flip-move default object is undefined

### ISSUE THREAD

**Issue by jmlsf** [thheller/shadow-cljs#194]
I have been trying to use https://github.com/joshwcomeau/react-flip-move in a shadow-cljs project, but I can't seem to get the default object to import.  Steps to repoduce:

Steps to repo flip-move issue

```sh
$ git clone https://github.com/shadow-cljs/quickstart-browser.git quickstart
$ cd quickstart
$ yarn add react-flip-move react@15.6.2 react-dom@15.6.2
$ yarn install
```
Change :dependencies to:
```clojure
 [[reagent "0.8.0-alpha2"]]
```

Change browser.cljs to:

```clojure
(ns starter.browser
  (:require [reagent.core :as reagent]
            ["react-flip-move" :as flip-move :default FlipMove]))

(defn start []
  (js/console.log "FlipMove :default = " FlipMove)
  (js/console.log "flip-move/default = " flip-move/default)
  (reagent/render [:div "hello world"]
                  (.getElementById js/document "app")))

(defn ^:export init []
  ;; init is called ONCE when the page loads
  ;; this is called in the index.html and must be exported
  ;; so it is available even in :advanced release builds
  (js/console.log "init")
  (start))

(defn stop []
  ;; stop is called before any code is reloaded
  ;; this is controlled by :before-load in the config
  (js/console.log "stop"))
```

On the console you'll get:

```
FlipMove :default =  undefined
browser.cljs:7 flip-move/default =  undefined
```

With normal cljs, I could get this to work by using foreign-libs and the UMD builds, and then accessing `js/FlipMove`.  Interestingly, `js/FlipMove` is undefined in the above code either, indicating that it just isn't getting loaded for some reason.

**Comment by jmlsf** [comment]
I'll give you one more totally useless piece of information: for some reason in my larger project that has way more dependencies, js/FlipMove is defined.  I can't figure out why.  But the default exports still do not work.

**Comment by thheller** [comment]
Good catch. I need to document this behavior.

`react-flip-move` is among those packages that pretends to have a ES6 default export but the default distribution is a CommonJS bundle so it doesn't.

I uploaded a working example here: https://github.com/thheller/issue-194

The only difference to your code is that on `:as` is used and not `:default`.

```clojure
(ns starter.browser
  (:require [reagent.core :as reagent]
            ["react-flip-move" :as FlipMove]))
```

This happens because `shadow-cljs` currently only uses the `main` entry of the package which usually points to a CommonJS file. `module` is convention for the ES6 entry but this is not yet used due to problems with a few packages that can't decide what they are doing. `material-ui-next` for example has a `module` entry for [index.es.js](https://unpkg.com/material-ui-next@1.0.0-beta.32/index.es.js) but this file imports relative files like [./AppBar](https://unpkg.com/material-ui-next@1.0.0-beta.32/AppBar/index.js) which then are CommonJS again.

In short the situation is a little messy.

In case of `react-flip-move` you can force it to use the ES6 version by either requiring it directly

```clojure
(ns starter.browser
  (:require [reagent.core :as reagent]
            ["react-flip-move/dist/react-flip-move.es" :default FlipMove]))
```

or by changing how it resolves in your build

```
:js-options
{:resolve
  {"react-flip-move" {:target :npm :require "react-flip-move/dist/react-flip-move.es"}}}
```

I will see if I can figure out a way to detect and warn about these "hybrid" packages.
